concepts/references/about.md

```diff
@@ -1,3 +1,100 @@
 # About
 
-TODO
+A [reference][reference] is a pointer: an address in memory where a value is stored.
+
+To understand the use of references it is helpful to briefly review ownership. There is only one owner of a value at a time, and when the owner
+goes out of scope the value will be dropped, as exampled below
+
+```rust
+fn log(msg: String) { // msg takes ownership of the value passed to it
+    let formatted_datetime: String;
+    // code  to format current datetime snipped
+    println!("{} at {}", msg, formatted_datetime);
+} // msg is dropped here
+
+pub fn main() {
+    let my_string = "Something happened".to_string();
+    log(my_string); // passing my_string transfers ownership of its value to msg
+    //println!("{:?}", my_string); // uncommenting this line will error because my_string's value has been dropped while owned by msg
+}
+```
+
+A [reference][borrow] represents a _borrow_ of an owned value. Instead of having `msg` own `my_string`'s value, we can have `msg` borrow it and
+then give it back to `my_string` at the end of `log`.
+
+To have `msg` borrow `my_string` we define `msg` as a reference and give it `my_string`'s address, not `my_string` itself, like so
+
+```rust
+fn log(msg: &String) { //msg is defined as a reference with an ampersand
+    let formatted_datetime: String;
+    // code  to format current datetime snipped
+    println!("{} at {}", msg, formatted_datetime);
+}
+
+pub fn main() {
+    let my_string = "Something happened".to_string();
+    log(&my_string); // my_string is passed as a reference with an ampersand
+    println!("{:?}", my_string);
+}
+```
+
+References are immutable, but they can be made mutable by appending `mut` to the ampersand. To access the value, the dereference operator (`*`)
```

Comment:
You haven't demonstrated that there can be many simultaneous immutable references. You could do a lot with something like [this](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=2cd4e96879021b6994db21e021a7e660):

```rust
fn is_constant(slice: &[u8]) -> bool {
    slice
        .first()
        .map(|first| slice.iter().all(|v| v == first))
        .unwrap_or_default()
}

fn is_linear(slice: &[u8]) -> bool {
    let diffs: Vec<_> = slice
        .windows(2)
        .map(|window| window[1] - window[0])
        .collect();
    is_constant(&diffs)
}

fn is_superlinear(slice: &[u8]) -> bool {
    let diffs: Vec<_> = slice
        .windows(2)
        .map(|window| window[1] - window[0])
        .collect();
    diffs.windows(2).all(|window| window[1] > window[0])
}

fn check_shapes(constant: &[u8], linear: &[u8], superlinear: &[u8]) -> (bool, bool, bool) {
    (
        is_constant(constant),
        is_linear(linear),
        is_superlinear(superlinear),
    )
}

fn main() {
    let fibonacci = vec![1, 1, 2, 3, 5, 8, 13];
    println!(
        "{:#?}",
        check_shapes(&fibonacci[..=1], &fibonacci[1..=3], &fibonacci[2..],)
    );
}
```

Reply:
Er, this note was meant to attach to line 61, not here, sorry.

Reply:
Yes, I felt lazy about not showing multi-immutables.  I like this basic structure of an example, but I wonder if the amount of code could be reduced to just snips for the is_X functions  so the volume of code isn't quite as overwhelming. Maybe use your line about leaving implementation as an exercise for the reader and returning a hard-coded bool instead?

Reply:
I'll leave that to external opinion. @efx?

My take is that these are three very simple, short functions. A hard-coded value plus a comment in each would save fewer than 10 lines of code. The joke works better with the code present, in my opinion.

Reply:
My take is that it deflects focus from the point, which I think is well-represented in `check_shapes`, I mean, when I first saw it I thought, wow! that's a lotta code!

I'm thinking something like

```rust
fn is_constant(slice: &[u8]) -> bool {
    true // implementation is left as an exercise for the reader
}
```
Or maybe to keep it with your implementation, but to put them at the bottom, with `check_shapes` on the top, so it starts with the point, something like

```rust
fn check_shapes(constant: &[u8], linear: &[u8], superlinear: &[u8]) -> (bool, bool, bool) {
    (
        is_constant(constant), // an immutable reference
        is_linear(linear), // another immutable reference, which is okay
        is_superlinear(superlinear), // okay, all three references are immutable
    )
}
```

Yeah, I think that might be best. It keeps your implementation but places the "point" first, at the top.

Reply:
I appreciate the topic of these example functions as it assembles teaching overlapping references nicely.

I do feel like it would be too much code to read initially. 
I love the in-depth "how does this work in a discrete code sample" but think they should go in a later section in the about. 

As an aside: we could consider establishing a "How does this actually look and work"? section for various `about.md` documents.

I was also wondering about a student who is reading this document but may not be familiar with iterator adapters or the functional style of programming. In that case, it seems like the implementations of `is_constant` may distract.

Reply:
@efx , how do you like it with the is_X shapes pushed to the bottom? Once I correct the comments might it be okay?

Reply:
Perhaps a comment can be put above the is_X functions something like "understanding the implementation of these functions is not necessary for this example but are provided should you be interested"?

Reply:
@bobahop Pushed to the bottom sounds great! 
The comment is nice too. We can always update as we get readers.